packages/turf-boolean-contains/index.js

```diff
@@ -0,0 +1,208 @@
+        if (deepEqual(MultiPoint.coordinates[i], Point.coordinates)) {
+            output = true;
+            break;
+        }
+    }
+    return output;
+}
+
+function isMultiPointInMultiPoint(MultiPoint1, MultiPoint2) {
+    var foundAMatch = 0;
+    for (var i = 0; i < MultiPoint2.coordinates.length; i++) {
+        var anyMatch = false;
+        for (var i2 = 0; i2 < MultiPoint1.coordinates.length; i2++) {
+            if (deepEqual(MultiPoint2.coordinates[i], MultiPoint1.coordinates[i2])) {
+                foundAMatch++;
+                anyMatch = true;
+                break;
+            }
+        }
+        if (!anyMatch) {
+            return false;
+        }
+    }
+    return foundAMatch > 0 && foundAMatch < MultiPoint1.coordinates.length;
+}
+
+// http://stackoverflow.com/a/11908158/1979085
+function isPointOnLine(LineString, Point) {
+    var output = false;
+    for (var i = 0; i < LineString.coordinates.length - 1; i++) {
+        if (isPointOnLineSegment(LineString.coordinates[i], LineString.coordinates[i + 1], Point.coordinates)) {
+            output = true;
+            break;
+        }
+    }
+    return output;
+}
+
+function isMultiPointOnLine(LineString, MultiPoint) {
+    var output = true;
+    for (var i = 0; i < MultiPoint.coordinates.length; i++) {
+        var pointIsOnLine = false;
+        for (var i2 = 0; i2 < LineString.coordinates.length - 1; i2++) {
+            if (isPointOnLineSegment(LineString.coordinates[i2], LineString.coordinates[i2 + 1], MultiPoint.coordinates[i])) {
+                pointIsOnLine = true;
+                break;
+            }
+        }
+        if (!pointIsOnLine) {
+            output = false;
+            break;
+        }
+    }
+    return output;
+}
+
+function isPointInPoly(Polygon, Point) {
+    return inside(Point, Polygon);
+}
+
+function isMultiPointInPoly(Polygon, MultiPoint) {
+    var output = true;
+    for (var i = 0; i < MultiPoint.coordinates.length; i++) {
+        var isInside = isPointInPoly(Polygon, helpers.point(MultiPoint.coordinates[1]));
+        if (!isInside) {
+            output = false;
+            break;
+        }
+    }
+    return output;
+}
+
+// TO DO - Work out how to check if line is in line (can potentially use line overlap module)
+// Also need to make sure lines are exactly the same, eg the second must be smaller than the first
+function isLineOnLine(LineString1, LineString2) {
+    var output = true;
+    var overlappingLine = lineOverlap(helpers.feature(LineString1), helpers.feature(LineString2)).features[0];
+    if (!overlappingLine) return false;
+    overlappingLine = getGeom(overlappingLine);
+    if (overlappingLine.coordinates[0] === LineString1.coordinates[0] &&
+        overlappingLine.coordinates[overlappingLine.coordinates.length - 1] === LineString1.coordinates[LineString1.coordinates.length - 1]) {
+        output = false;
+    }
+    return output;
+}
+
+function isLineInPoly(Polygon, Linestring) {
```

Comment:
👍 Polygons with holes will be hard to implement, @rowanwins I think at first lets just get it working with only outer rings and afterwards we work on the inner ring validation (donuts).